test/test_nestedtensor.py

```diff
@@ -3342,6 +3342,39 @@ def test_unbind(self, device):
             for i, t in enumerate(out):
                 self.assertEqual(t, tensor_list[i])
 
+    @torch._dynamo.config.patch(suppress_errors=True)
+    @dtypes(torch.float, torch.double, torch.half)
+    @parametrize("requires_grad", [False, True])
+    def test_factory_functions(self, device, dtype, requires_grad):
+        for tensor_list in self._get_example_tensor_lists():
+            kwargs = {
+                "device": device,
+                "dtype": dtype,
+                "requires_grad": requires_grad,
+            }
+            nt = torch.nested.nested_tensor(
+                tensor_list,
+                layout=torch.jagged,
+                **kwargs
+            )
+            zeros = torch.zeros(nt.shape, **kwargs)
+            ones = torch.ones(nt.shape, **kwargs)
+            empty = torch.empty(nt.shape, **kwargs)
+            full = torch.full(nt.shape, 2, **kwargs)
```

Comment:
can we also get a test where parts of `nt.shape` are taken alongside other shape parts? I think that'll be a core use case for this stuff in model code. I had to do a bunch of gymnastics in SAM to get around this not existing then